### DISCUSSION TITLE
CMD Reiter im Iobroker fehlt

### DISCUSSION THREAD

**Discussion by FloW-80** [tbnobody/OpenDTU#462] [Q&A]
### What happened?

Bei mir sind im Iobroker unter Objekte - mqtt - solar und dann der jeweilige Inverter nur 0,1,2,3,4, Device und Status vorhanden,  CMD fehlt. Ich hatte letzte Woche erst die DTU auf die neueste Version gebracht.

### To Reproduce Bug

-

### Expected Behavior

-

### Install Method

Pre-Compiled binary from GitHub

### What git-hash/version of OpenDTU?

9bfc704

**Comment by tbnobody** [comment]
Ich weiß nicht was ioBroker als Tabs darstellt. Ich vermute das sind die verschiedenen Topics die ioBroker in der Vergangenheit schon einmal gesehen hat. Wenn du noch nichts an eine Topic geschickt hast wird die dort auch nicht auftauchen.

**Comment by JauntyJosef** [comment]
Genau so ist es. Wenn du noch nie was an das Topic cmd geschickt hast, dann siehst du im ioBroker auch nichts.

**Comment by Dodger2022** [comment]
Hallo,
ich muss hier nochmal genau nachfragen, da ich das gleiche "Problem" habe.

Ich versuche seit Tagen diese Topics zum laufen zu bringen damit ich die Max-Leistung über IObroker steuern kann.
(soll mal eine Nacheinspeisung von einem extern Batteriesystem werden an einem freien Eingang an einem HM-600)

Ich habe versuche über den MQTT-Explorer die Topics anzusprechen und den HM600 zu regeln. 
Ich sehe auch im Explorer die Meldungen Name;Device;status;0;1;2 usw mit ihren Inhalten.
Wenn ich unter Puplish/Topic etwas eintrage und puplishe legt IOBroker diese Einträge auch an, aber der Hoymiles setzt es nicht um oder schaltet ab (Statusleiste wird gelb) oder setzt Current Limit auf 0W/%.
![MQTT2](https://user-images.githubusercontent.com/119224337/208450208-2082d461-1cf6-41a2-83d3-fc416b1f543e.PNG)


(zb : OpenDTU/serialXXXXXXXserial/cmd/limit_nonpersistent_relative/50% ) 
       
Oder ist nur die Schreibweise falsch?

Interessant finde ich auch, das wenn ich OpenDTU neu starte tauchen im MQTT-Log des IOBrokers die CMD-Topics kurz auf, werden aber in der Objektliste nicht angezeigt/angelegt.
![MQTT1](https://user-images.githubusercontent.com/119224337/208450967-c6594c71-1144-4459-9d36-71dd540a04f4.PNG)


Sorry wenn ich mich doof anstelle aber muss mich mit MQTT noch vertraut machen.

**Comment by tbnobody** [comment]
> (zb : OpenDTU/serialXXXXXXXserial/cmd/limit_nonpersistent_relative/50% )

Das ist hier ist falsch
Bei Topic muss `OpenDTU/serialXXXXXXXserial/cmd/limit_nonpersistent_relative` stehen. Der Wert (Eingabefeld darunter im MQTT Explorer) muss `50` sein.

**Comment by romew137** [comment]
[Klick mich](https://github.com/tbnobody/OpenDTU/issues/35)
Lest euch mal durch, was Hubsi5 am 14 Oktober geschrieben hat.
Hatte das selbe Problem und habe es damit behoben bekommen.

**Comment by tbnobody** [comment]
Ich verschiebe das mal in die Diskussionen weil es kein Issue von OpenDTU ist.

**Comment by Dodger2022** [comment]
Hallo,
bei mir läuft das ganze jetzt. Danke an alle die geholfen haben.
Ich beschreibe mal was ich geändert habe, vielleicht hilfts dem ein oder anderern.

Erstmal habe ich die MQTT Instance im IOBroker neu installiert das ich diese scheinbar kaputt konfiguriert habe. Hier meine jetzigen Einstellungen.

![3](https://user-images.githubusercontent.com/119224337/209436535-779c2c4f-6763-4d69-a989-cc261723850e.PNG)

Das Feld "Nachrichten ohne retain-Flag" senden musste ich anhaken da sonst der Hoymiles nicht reagierte.

Dann mit MQTT Broker das cmd Topic puplishen . Daraufhin taucht im IOBroker das Objekt /cmd/ auf.

![1](https://user-images.githubusercontent.com/119224337/209436584-d9b80099-711d-44eb-95d4-492d47c67bb9.PNG)


![2](https://user-images.githubusercontent.com/119224337/209436665-587bfdf2-11e4-4f20-a40f-e67b9d6b1bc1.PNG)

Jetzt kann ich einfach den Wert im Objekt (IOBroker) ändern und bestätigen und der Hoymiles setzt das mit ca. 3 Sek verzögerung um.

Nochmals danke an alle Helfer
